src/types.ts

```diff
@@ -214,7 +265,15 @@ export interface PostHogConfig {
     feature_flag_request_timeout_ms: number
     get_device_id: (uuid: string) => string
     name: string
+    /**
+     * this is a read-only function that can be used to react to event capture
+     */
     _onCapture: (eventName: string, eventData: CaptureResult) => void
+    /**
+     * This function - if provided - is called immediately before sending data to the server.
+     * It allows you to edit data before it is sent, or choose not to send it all.
+     */
+    before_send: (cr: CaptureResult) => CaptureResult | null
```

Comment:
Should we deprecate `_onCapture`? I mean before send does the same, if you don't want to mutate anything, just dont.

Reply:
I guess we should, yes. 

I was leaning away to let us test more widely but that's overly cautious 👍

Reply:
I mean we don't need to remove anything, and an @deprecated annotation is just an annotation anyway, but we can do the annotation later once we are more confident as well, that's fine.

Reply:
🤣 turns out deprecated is a github user 🤣 

👋 

----

deprecated _onCapture and moved all the test usages over to before_send